crates/turbo-tasks-memory/src/task.rs

```diff
@@ -92,15 +106,160 @@ pub struct Task {
     /// The type of the task
     ty: TaskType,
     /// The mutable state of the task
-    state: RwLock<TaskState>,
+    /// Unset state is equal to a Dirty task that has not been executed yet
+    state: RwLock<TaskMetaState>,
+}
+
+enum TaskMetaState {
```

Comment:
Can you move this enum and its impls into its own module? This file is already very large.